packages/flutter/test/widgets/form_test.dart

```diff
@@ -886,4 +886,40 @@ void main() {
     expect(fieldValue, '123456');
     expect(formKey.currentState.validate(), isFalse);
   });
+
+  testWidgets('autovalidate parameter is passed into class property', (WidgetTester tester) async {
+    String errorText(String value) => '$value/error';
+    const ObjectKey widgetKey = ObjectKey('key');
+
+    Widget builder() {
+      return MaterialApp(
+        home: MediaQuery(
+          data: const MediaQueryData(devicePixelRatio: 1.0),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: Center(
+              child: Material(
+                child: FormField<String>(
+                  key: widgetKey,
+                  initialValue: 'foo',
+                  autovalidate: true,
+                  builder: (FormFieldState<String> state) {
+                    return Container();
+                  },
+                  validator: errorText,
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(builder());
+
+    final Finder formFieldFinder = find.byKey(widgetKey);
+    final FormField<String> formField = tester.widget(formFieldFinder);
+    expect(formField.autovalidate, isTrue);
+    expect(formField.autovalidateMode, equals(AutovalidateMode.always));
```

Comment:
I think to be comprehensive you should test when autovalidate is false, and also do the same two tests for Form.

You could make `builder` take a parameter for `autovalidate` and build it a second time set to false.

Reply:
Yes, this test title also needs a improvement!